rewrite-yaml/src/main/java/org/openrewrite/yaml/MergeYaml.java

```diff
@@ -80,6 +80,11 @@ public class MergeYaml extends Recipe {
     @Nullable
     String insertProperty;
 
+    // Although the add-new-key-when-key-does-not-match behaviour was intentionally programmed, after more thought, we do not want it anymore,
+    // as the user of this recipe can achieve the very same result by using the root key (which is $).
+    // This boolean is intended for imperative recipes that rely on this old behavior.
+    boolean createNewKeys;
```

Comment:
I am starting to doubt this decision. Changing this behaviour has impact on other recipes (see my changes for the CopyValue recipe). It could very well be true customer recipes also use the MergeYaml recipe as a base. So I opt to go for a Boolean @Option instead, which by default uses the current behaviour.

Reply:
Perhaps this could be covered by a precondition rather than adding another option to this recipe? I don't know what precondition recipes we have for yaml though

Reply:
No, because whether you want to add a key or not is not something you can infer from the incoming/existing yaml (or any other object). What matters is the intent of the user's use of the recipe..